### DISCUSSION TITLE
Logging in with user account does not work on Linux

### DISCUSSION THREAD

**Discussion by georgesuba** [datenstrom/community#163] [Report a bug] [answered]
Hi.. I've created user account according to docs in wiki but when I press edit the page with correct email and passwd then page just reload.. and nothing happen. what permissions should be on files and folders?

I got for folders: 755
for files: 644
apache 2.4
congif as provided with downloaded yellow

Thanks.

EDIT: additionally when login window is closed and any page clicked on menu, page is loaded with login window over top of it.. this behaviour occurs in both in latest chrome and ff.

**Comment by markseuffert** [comment]
Can you login and edit a page in the [Yellow demo](https://datenstrom.se/yellow/demo/)? Does that work for you?

**Comment by georgesuba** [comment]
on demo page yes.. but not on my server

**Comment by markseuffert** [comment]
Which Yellow version are you using?

**Comment by georgesuba** [comment]
@markseu 

Okay my bad.. everything works on production server. I think its problem on my dev server which sits on localhost:8080.. have no idea why it is doing that as dev server setup is exactly same as production server. I will try to figure this out so I can develop and test themes for this great cms.

**Comment by markseuffert** [comment]
:smile:

Good to hear... but strange that it doesn't work on your development server. I have a similar workflow, I develop on my regular computer (Macbook) and then upload it later to the production web server (Linux). Can I ask something, what ~~web server and~~ operating system are you using on localhost?

**Comment by georgesuba** [comment]
Sure.. its debian 8 - virtualmin installed - local domains and subdomains running in virtualbox.. :) everything works just this cms wont .. I mean cms works fine just login. I think js somehow does not like different ports :D

**Comment by georgesuba** [comment]
I am really thankful for this cms..  is there any skeleton theme like empty one?

**Comment by markseuffert** [comment]
Should be working now, setting cookies was wrong for localhost with port number. Oops my bad.

Can you please test latest webinterface plugin, does it work for you?

**Comment by georgesuba** [comment]
it didn't work for me but I see correction :) It did not worked as I am using dev.dev as localhost instead of localhost so I am sure it will work on localhost!

As I said I have server in virtualbox redirected in my linux os as localhost.. I have sort of  local domains like dev.dev or whatever I will name it  so I've changed two lines:

`if(preg_match("/^localhost(:\d+)?$/", $serverName)) $serverName = false;`
to
`if(preg_match("/^dev.dev(:\d+)?$/", $serverName)) $serverName = false;`

and it works - Thanks. Its my funny local made up domain so I've just tmp changed this there.. I think cookies are now correct.

**Comment by markseuffert** [comment]
Here's another attempt, setting cookies should now work better with port numbers. Give it a try please.

**Comment by georgesuba** [comment]
this one does not work at all.. I would rather retrieve that file back to older version - the one where was those two lines: `if(preg_match("/^localhost(:\d+)?$/", $serverName)) $serverName = false;` - that was working on custom localhost domains fine when edited towards user needs..

**Comment by markseuffert** [comment]
@wunderfeyd Can you have a look please?

**Comment by wunderfeyd** [comment, marked as answer]
@georgesuba Could you update the [`system/plugins/webinterface.php`](https://github.com/datenstrom/yellow/blob/master/system/plugins/webinterface.php) to version 0.6.3 please. Please delete your cookies for the website/your installation and test again.

**Comment by georgesuba** [comment]
morning guys.. @wunderfeyd I can confirm that 0.6.3 version worked right a way on localhost:custom-ports and localhost-custom-domain:custom-ports (dev.dev:8080) without any modifications to the file. It also works on normal production server and domain right a way - this is great. Thank you.

EDIT: it now works on custom subdomains on localhost and custom ports as well (yellow.dev.dev:8080) :)

**Comment by markseuffert** [comment]
It also works with different operating systems, we tested it again on iOS, Linux, Mac, Windows.

Thank you guys for your help, much appreciated. Please open a new issue if there are more questions.
